backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/types/codec/impl/AbstractElasticsearchFieldCodec.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.search.backend.elasticsearch.types.codec.impl;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+
+public abstract class AbstractElasticsearchFieldCodec<T> implements ElasticsearchFieldCodec<T> {
+
+	protected final Gson gson;
+
+	public AbstractElasticsearchFieldCodec(Gson gson) {
+		this.gson = gson;
+	}
+
+	@Override
+	public JsonElement fromJsonStringToElement(String value) {
+		return gson.fromJson( value, JsonElement.class );
+	}
+
+	@Override
+	public String fromJsonElementToString(JsonElement value) {
+		if ( value == null || value.isJsonNull() ) {
+			return null;
+		}
+		return gson.toJson( value );
+	}
```

Comment:
oooooooook 😄 
So something like this I guess ... and then a lot of noise for the codec change (so may want to ignore that) though some codecs now are not singletons anymore